core/vm/environment.libevm.go

```diff
@@ -134,13 +152,17 @@ func (e *environment) callContract(typ CallType, addr common.Address, input []by
 
 		startGas := gas
 		defer func() {
-			t.CaptureEnd(retData, startGas-retGas, retErr)
+			t.CaptureEnd(retData, startGas-e.Gas(), retErr)
 		}()
 	}
 
 	switch typ {
 	case Call:
-		return e.evm.Call(caller, addr, input, gas, value)
+		ret, returnGas, err := e.evm.Call(caller, addr, input, gas, value)
+		if err := e.refundGas(returnGas); err != nil {
+			return nil, err
```

Comment:
add context to the error so we know where it's coming from

```suggestion
			return nil, fmt.Errorf("refunding gas: %w", err)
```

Reply:
This can only be `ErrUintGasOverflow` so it's implicit in the error. See other comments re not wrapping the upstream errors.